component/src/main/java/io/siddhi/extension/io/grpc/sink/AbstractGrpcSink.java

```diff
@@ -0,0 +1,195 @@
+    protected boolean isDefaultMode = false;
+    protected String url;
+    protected String streamID;
+    protected String address;
+    protected EventServiceGrpc.EventServiceFutureStub futureStub;
+    protected Option headersOption;
+    protected ManagedChannelBuilder managedChannelBuilder;
+    protected long channelTerminationWaitingTime;
+
+    /**
+     * Returns the list of classes which this sink can consume.
+     * Based on the type of the sink, it may be limited to being able to publish specific type of classes.
+     * For example, a sink of type file can only write objects of type String .
+     * @return array of supported classes , if extension can support of any types of classes
+     * then return empty array .
+     */
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{com.google.protobuf.GeneratedMessageV3.class, String.class};
+        // in default case json mapper will inject String. In custom gRPC service
+        // case protobuf mapper will inject gRPC message class
+    }
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    /**
+     * Returns a list of supported dynamic options (that means for each event value of the option can change) by
+     * the transport
+     *
+     * @return the list of supported dynamic option keys
+     */
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[]{GrpcConstants.HEADERS};
+    }
+
+    /**
+     * The initialization method for {@link Sink}, will be called before other methods. It used to validate
+     * all configurations and to get initial values.
+     * @param streamDefinition  containing stream definition bind to the {@link Sink}
+     * @param optionHolder            Option holder containing static and dynamic configuration related
+     *                                to the {@link Sink}
+     * @param configReader        to read the sink related system configuration.
+     * @param siddhiAppContext        the context of the {@link io.siddhi.query.api.SiddhiApp} used to
+     */
+    @Override
+    protected StateFactory init(StreamDefinition streamDefinition, OptionHolder optionHolder, ConfigReader configReader,
+                                SiddhiAppContext siddhiAppContext) {
+        this.siddhiAppContext = siddhiAppContext;
+        this.url = optionHolder.validateAndGetOption(GrpcConstants.PUBLISHER_URL).getValue();
+        this.streamID = siddhiAppContext.getName() + GrpcConstants.PORT_HOST_SEPARATOR + streamDefinition.getId();
+        if (optionHolder.isOptionExists(GrpcConstants.HEADERS)) {
+            this.headersOption = optionHolder.validateAndGetOption(GrpcConstants.HEADERS);
+        }
+        if (!url.substring(0, 4).equalsIgnoreCase(GrpcConstants.GRPC_PROTOCOL_NAME)) {
+            throw new SiddhiAppValidationException(streamID + "The url must begin with \"" +
```

Comment:
Add Siddhi App name and streamID for all the exceptions, also space is needed here `streamID + "The url`. 
Formate the text correctly !!!!

Reply:
Check all the exceptions and add Siddhi App name and streamID and formate the text correctly !!!!

Reply:
I have added siddhi app name in the streamID. will remove the siddhi app name part and add seperately in each error